core-bundle/src/Resources/public/core.js

```diff
@@ -992,10 +993,50 @@ var Backend =
 	},
 
 	/**
-	 * Get the current scroll offset and store it in a cookie
+	 * Store the current scroll offset in localstorage
 	 */
 	getScrollOffset: function() {
-		document.cookie = "BE_PAGE_OFFSET=" + window.getScroll().y + "; path=" + (Contao.path || '/');
+		window.localStorage.setItem('contao_backend_offset', parseInt(window.getScroll().y, 10) + window.location);
+	},
+
+	/**
+	 * Scrolls to the current offset if it was defined and gives the header the "down" CSS class.
+	 * It also removes the legacy BE_PAGE_OFFSET cookie.
+	 */
+	initScrollOffset: function() {
+		// Make sure we always kill the legacy cookie here
+		// That way it can be sent by the server but it wont be resent by the
+		// client in the next request
+		Cookie.dispose('BE_PAGE_OFFSET');
+
+		var item = window.localStorage.getItem('contao_backend_offset'), additionalOffset = 0;
+
+		if (!item) {
+			return;
+		}
+
+		// Remove local storage item again
+		window.localStorage.removeItem('contao_backend_offset');
+
+		var chunks = item.split(/^(\d*)/);
```

Comment:
I'm storing the offset plus the current window location, so the item contains something like `768https://......./contao?....`. Why do I store the current location? Well, it's to replace the difference between the `save` and the `saveNcreate`, `saveNclose` etc. buttons. Just a lot smarter 😄 The old code called `System::setCookie('BE_PAGE_OFFSET', 0, 0);` for all the buttons except for the `save` button which is the one where you stay on the same page. This makes sense, if you save (e.g. using keyboard shortcuts) and stay on the same page, you want to jump back to the location where you've last edited but you don't want to jump there if you are on a different page (which you are when you `saveNclose` etc.). So I thought about adding some data attribute to the buttons but then I noticed that it only ever makes sense not to reset, when you are on the same page. So I'm storing the `window.location` along with the offset and if they do not match, I'm not jumping there which is bascially exactly the same functionality as we have now but it works without having to configure anything at all.